tests/test-strict.lua

```diff
@@ -1,3 +1,4 @@
+warn = warn or true-- declare 'warn' as a global, to be added in compat module
```

Comment:
I'm not up to speed on this test or what the issue is this solves, but it doesn't smell quite right. Not a huge concern since test migration is still in process anyway and this is in the legacy section.

Reply:
`compat` is loaded after `strict` below. And this makes `strict` trip an error when `compat` tries to set a new global `warn`

Reply:
this probably warrants a warning message in the changelog, along with the deprecation warnings.

Reply:
Isn't a true value in the global namespace

1) where a function belongs a type casting no-no?
2) going to block the compat layer from even noticing that it needs to add its shim?

Reply:
yep, you're right. Updated to load compat before strict.